pennylane/ops/qutrit/parametric_ops.py

```diff
@@ -306,10 +309,154 @@ def compute_matrix(theta, subspace=(0, 1)):  # pylint: disable=arguments-differ
+    basis states the operation applies to.
+
+    The construction of this operator is based on section 3 of
+    `Di et al. (2012) <https://arxiv.org/abs/1105.5485>`_.
+
+    .. math:: TR_z^{jk}(\phi) = \exp(-i\phi\sigma_z^{jk}/2),
+                \sigma_z^{jk} = |j\rangle\langle j| - |k\rangle\langle k|,
+                j, k \in \{0, 1, 2\}, j \neq k
+
+    **Details:**
+
+    * Number of wires: 1
+    * Number of parameters: 1
+    * Number of dimensions per parameter: (0,)
+
+    Args:
+        phi (float): rotation angle :math:`\phi`
+        wires (Sequence[int] or int): the wire the operation acts on
+        subspace (Sequence[int]): the 2D subspace on which to apply operation
+        do_queue (bool): Indicates whether the operator should be
+            immediately pushed into the Operator queue (optional)
+        id (str or None): String representing the operation (optional)
+
+    **Example**
+
+    The specified subspace will determine which basis states the operation actually
+    applies to:
+
+    >>> qml.TRZ(0.5, wires=0, subspace=[0, 1]).matrix()
+    array([[0.96891242-0.24740396j, 0.        +0.j        , 0.        +0.j        ],
+           [0.        +0.j        , 0.96891242+0.24740396j, 0.        +0.j        ],
+           [0.        +0.j        , 0.        +0.j        , 1.        +0.j        ]])
+
+    >>> qml.TRZ(0.5, wires=0, subspace=[0, 2]).matrix()
+    array([[0.96891242-0.24740396j, 0.        +0.j        , 0.        +0.j        ],
+           [0.        +0.j        , 1.        +0.j        , 0.        +0.j        ],
+           [0.        +0.j        , 0.        +0.j        , 0.96891242+0.24740396j]])
+
+    >>> qml.TRZ(0.5, wires=0, subspace=[1, 2]).matrix()
+    array([[1.        +0.j        , 0.        +0.j        , 0.        +0.j        ],
+           [0.        +0.j        , 0.96891242-0.24740396j, 0.        +0.j        ],
+           [0.        +0.j        , 0.        +0.j        , 0.96891242+0.24740396j]])
+    """
+    num_wires = 1
+    num_params = 1
+    """int: Number of trainable parameters that the operator depends on."""
+
+    ndim_params = (0,)
+    """tuple[int]: Number of dimensions per trainable parameter that the operator depends on."""
+
+    grad_method = "A"
+    parameter_frequencies = [(0.5, 1)]
+
+    def generator(self):
+        if self.subspace == (0, 1):
+            return qml.s_prod(-0.5, qml.GellMann(wires=self.wires, index=3))
+
+        if self.subspace == (0, 2):
+            coeffs = [-0.25, -0.25 * np.sqrt(3)]
+            obs = [qml.GellMann(wires=self.wires, index=3), qml.GellMann(wires=self.wires, index=8)]
+            return qml.dot(coeffs, obs)
+
+        coeffs = [-0.25 * np.sqrt(3), 0.25]
+        obs = [qml.GellMann(wires=self.wires, index=8), qml.GellMann(wires=self.wires, index=3)]
+        return qml.dot(coeffs, obs)
+
+    def __init__(self, phi, wires, subspace=(0, 1), do_queue=True, id=None):
+        self._subspace = self.validate_subspace(subspace)
```

Comment:
In `TRX` it is `Operation.validate_subspace`, any particular reason for the difference?